langchain/chains/query_constructor/prompt.py

```diff
@@ -0,0 +1,137 @@
+# flake8: noqa
+from langchain import PromptTemplate
+
+song_data_source = """\
+```json
+{
+    content: "Lyrics of a song",
+    attributes: {
+        "artist": {
+            "type": "string",
+            "description": "Name of the song artist"
+        },
+        "length": {
+            "type": "integer",
+            "description": "Length of the song in seconds"
+        },
+        "genre": {
+            "type": "string",
+            "description": "The song genre, one of \"pop\", \"rock\" or \"rap\""
+        }
+    }
+}
+```\
+""".replace(
+    "{", "{{"
+).replace(
+    "}", "}}"
+)
+
+full_answer = """\
+```json
+{{
+    "query": "teenager love",
+    "filter": "and(or(eq(\\"artist\\", \\"Taylor Swift\\"), eq(\\"artist\\", \\"Katy Perry\\")), \
+lt(\\"length\\", 180), eq(\\"genre\\", \\"pop\\"))"
+}}"""
+
+no_filter_answer = """\
+```json
+{{
+    "query": "",
+    "filter": "NO_FILTER"
+}}
+```\
+"""
+
+default_examples = [
```

Comment:
nit: If tuple works by default instead of list with downstream use case, we could use a tuple here to make the top level container immutable

Reply:
think examples prompt template expects a list, but maybe something to refactor. could also just convert to list when passing in